### PR TITLE
Improve documentation

### PR DESCRIPTION
**Pull request by Xowap**
As I tried to use this project, I had a few hurdles on the path and so I've tried to improve the experience for future users by modernizing the documentation:

- Create a Markdown version of the doc (the plain-text version is kept because autotools require it, but it's just a converted version of the Markdown one)
- More detailed explanations on the compilation process
- Fix a compilation warning

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+vary.
+
+
+## Who Contributed?
+
+The main author is Pat Padgett. But none of the contact info left work anymore,
+so a name is all we have.
+
+Bryan Chan created this repository and tweaked the code a little bit. Then
+Rémy Sanchez improved the documentation.
```
Comment (bryanpkc) [inline README.md:130]: Would you mind removing this section, and adding your name (and email address) to the AUTHORS file instead? I think the AUTHORS file should just be a flat list that looks like this:
```
Pat Padgett <agroman@agroman.net>
Bryan Chan <bryanpkc@gmail.com>
Rémy Sanchez <...>
```

File: README.md [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+# For Red-Hat-based distributions (CentOS, Fedora, ...)
+sudo yum groupinstall 'Development tools'
+```
+
+You will also need the source code. If you don't already have it, please
+clone it:
+
+```bash
+git clone https://github.com/bryanpkc/corkscrew.git
+```
```
Comment (bryanpkc) [inline README.md]: This bit on cloning the source code seems a little redundant. If the reader is already looking at this file, they already have the source code. :smile: I suggest deleting it.

File: README.md [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+Corkscrew's primary distribution site was 
+[agroman.net/corkscrew](https://web.archive.org/web/20170510154150/http://agroman.net/corkscrew/), 
+however it seems that the site went down and this repository is here to keep
+the code available. The new location is then 
+[github.com/bryanpkc/corkscrew](https://github.com/bryanpkc/corkscrew).
+
+
+## How Do I Install It?
+
+First you need to install development tools:
```
Comment (bryanpkc) [inline README.md:40]: Official source releases would have the autoconf stuff pre-generated, therefore typically all people need to do is following the INSTALL instructions. I suggest replacing this line with the following:
```markdown
If the `configure` shell script already exists in the source distribution, simply run:

    ./configure
    make install

Refer to the INSTALL file for more information.

If the `configure` shell script does not exist, you have probably just cloned the source code from GitHub.
You will need the following dependencies to build the project with GNU autoconf.
```

File: corkscrew.c [hunk 4]
```diff
@@ -202,7 +203,13 @@ char *argv[];
 				fprintf(stderr, "Error opening %s: %s\n", argv[5], strerror(errno));
 				exit(-1);
 			} else {
-				fscanf(fp, "%4095s", line);
+				line_ok = fscanf(fp, "%4095s", line);
```
Comment (bryanpkc) [inline corkscrew.c]: I would do this in an `else if` and get rid of the `line_ok` variable:
```c
} else if (!fscanf(fp, "%4095s", line)) {
```

**Review by bryanpkc — CHANGES_REQUESTED** [review]
Thank you for the pull request.

File: README.md (hunk 1 above)
Comment (Xowap) [reply README.md:130]: Yup makes sense

File: README.md (hunk 2 above)
Comment (Xowap) [reply README.md]: That is a reasonable point, I hesitated before putting it. Will remove.

File: README.md (hunk 3 above)
Comment (Xowap) [reply README.md:40]: Well, there is no official source release right?

Maybe we can keep it this way and then put it in a different section when official releases are being reinstated?

Comment (bryanpkc) [reply README.md:40]: Fair enough.

**Comment by Xowap** [comment]
I've done the updates, what do you think?

**Review by bryanpkc — APPROVED** [review]

**Comment by bryanpkc** [comment]
Thank you so much!

**Comment by Xowap** [comment]
Thanks to you :smile:

**Pull request merged**